aea/configurations/data_types.py

```diff
@@ -658,6 +588,10 @@ def to_uri_path(self) -> str:
         """
         return f"{str(self.package_type)}/{self.author}/{self.name}/{self.version}"
 
+    def without_hash(self,) -> "PackageId":
+        """Returns PackageId object without hash"""
+        return PackageId(self.package_type, self.public_id.without_hash())
```

Comment:
or we do the reverse and have `with_hash` so to keep compatibility automatically?